docroot/modules/custom/sitenow_events/sitenow_events.module

```diff
@@ -6,169 +6,69 @@
-      ->get('sitenow_events.filters_endpoint'),
-    \Drupal::config('sitenow_events.settings')
-      ->get('sitenow_events.keywords_endpoint'),
-  ];
-  foreach ($endpoints as $endpoint) {
-    try {
-      $response = \Drupal::httpClient()
-        ->get($endpoint, ['headers' => ['Accept' => 'text/plain']]);
-      $data = (string) $response->getBody();
-      if (empty($data)) {
-        \Drupal::messenger()->addError(t('Empty response.'));
-      }
-      else {
-        $results = json_decode($data, TRUE);
+function sitenow_events_preprocess_block(&$variables) {
+  if ($variables['elements']['#plugin_id'] == 'inline_block:uiowa_events') {
+    $variables['#attached']['library'][] = 'uids_base/events';
+
+    /** @var \Drupal\Block\Entity\Block $block */
+    $block = $variables['elements']['content']['#block_content'];
+    $start = date('d-m-Y');
+
+    $events = sitenow_events_load([
+      'display_id' => 'events',
+      'filters[department]' => $block->get('field_uiowa_events_department')->value,
+      'filters[audiences]' => $block->get('field_uiowa_events_audiences')->value,
+      'filters[interests]' => $block->get('field_uiowa_events_interests')->value,
+      'filters[types]' => $block->get('field_uiowa_events_types')->value,
+      'filters[keywords]' => $block->get('field_uiowa_events_keywords')->value,
+      'filters[startdate][value][date]' => date('m-d-Y', strtotime($start)),
+      'filters[enddate][value][date]' => date('m-d-Y', strtotime($start . '+365 days')),
+      'page' => 0,
+      'items_per_page' => $block->get('field_uiowa_events_results')->value ?? 50,
+    ]);
+
+    if (!empty($events['events'])) {
+      foreach ($events['events'] as $e) {
+        $variables['content'][] = [
+          '#theme' => 'sitenow_events_teaser',
+          '#event' => $e,
+        ];
       }
     }
-    catch (RequestException $e) {
-      watchdog_exception('sitenow_events', $e);
-    }
-    if (!empty($results)) {
-      $forest = [];
-      // Build a parent/children tree for each filter.
-      foreach ($results as $filter => $values) {
-        $forest[$filter] = _build_options_tree($values);
-      }
-      // Output a formatted options array for each filter.
-      foreach ($forest as $filter => $tree) {
-        $options[$filter] = _build_options_list($tree);
-      }
+    else {
+      $variables['content'] = [
+        '#markup' => '<div class="events-empty"><p>There are currently no events to display.</p></div>',
+      ];
     }
-  }
-  // Sort the options.
-  ksort($options);
-  return $options;
-}
 
-/**
- * Helper function to build the options tree.
- *
- * @param array $data
- *   Array of data to be sorted into a tree.
- * @param int $parent
- *   Stores the current id.
- *
- * @return array
- *   Returns an associative array options tree.
- */
-function _build_options_tree(array $data, $parent = 0) {
-  $tree = [];
-  foreach ($data as $d) {
-    if ($d['parent_id'] == $parent) {
```

Comment:
I think this loose comparison is a recursion trap but changing it breaks all the filters. It currently is equating null and 0.

Reply:
Working on this.